### PR TITLE
Make SSD character fall asleep

### LINKED ISSUE #21906: Feature request: SSD characters should really fall asleep

### PR DESCRIPTION
**Pull request by lzk228**
## About the PR

If entity is SSD, it got asleep with force (cannot be awaked)
When entity no longer SSD, it can be awaked
Resolves #21906 

## Why / Balance

Cool

## Technical details

no

## Media

https://github.com/space-wizards/space-station-14/assets/124214523/2568d015-849c-456e-858a-547c0bb6ef3a

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- tweak: SSD character now will fall asleep until they are no longer SSD.

### CODE REVIEW

**Comment by lzk228** [comment]
test are failing because of changes, im powerless here

**Comment by Tayrtahn** [comment]
> test are failing because of changes, im powerless here

Add a check to make sure the entity isn't terminating before adding the component.

**Comment by ps3moira** [comment]
Only downside is if the server sucks and you're disconnected your character drops whatever they're holding when they go to sleep, but this is awesome!

**Comment by VigersRay** [comment]
Omg, he is died

File: Content.Shared/SSDIndicator/SSDIndicatorSystem.cs
Comment (Tayrtahn) [inline Content.Shared/SSDIndicator/SSDIndicatorSystem.cs:27]:
```suggestion
        if (!TerminatingOrDeleted(uid))
            EnsureComp<ForcedSleepingComponent>(uid);
```

**Comment by Partmedia** [comment]
Thanks for the PR. Per the discussion on the original "Show SSD Indicator" CVar (I can link those if you can't find them), this behavior should probably also be controlled by the same CVar, i.e. not fall asleep if the CVar is disabled.

**Comment by lzk228** [comment]
> Thanks for the PR. Per the discussion on the original "Show SSD Indicator" CVar (I can link those if you can't find them), this behavior should probably also be controlled by the same CVar, i.e. not fall asleep if the CVar is disabled.

Tried to add check for the cvar but on ghost i got
```
[ERRO] con: localhost@JoeGenero: ExecuteError - ghost :
Robust.Shared.Configuration.InvalidConfigurationException: Trying to get unregistered variable 'ic.show_ssd_indicator'
```
for server and
```
[INFO] CON: > ghost
[INFO] CON: There was an error while executing the command: Robust.Shared.Configuration.InvalidConfigurationException: Trying to get unregistered variable 'ic.show_ssd_indicator'
```
for client

**Comment by Anzarot121** [comment]
revenant buff real

**Comment by ps3moira** [comment]
is this still being worked on?

**Comment by lzk228** [comment]
> is this still being worked on?

its working, i just wait for the issue to be resolved, because i have no clue how to resolve it and as Tayrtahn said its not my problem, i just add the code that cause the issue to happen

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Needs cvar for servers that may want it + the fact it has ingame implications due to revenant.

**Comment by SlamBamActionman** [comment]
Hello! The PR was put up to a maintainer vote which concluded with the following result: Merge, but with changes.

Maintainers think it's a neat idea, but that immediately falling asleep would be too punishing. This is due to the existence of revenants, who can kill a character who is sleeping, and also an antag DCing and finding themselves asleep even upon an immediate reconnect. 

The PR has maintainer approval if there is a timer before the sleep happens. This delay should be controlled by a cvar so that it's easily adjustable on a server-to-server basis.

An exact time wasn't settled upon but there were suggestions ranging from 10 to 20 minutes.

**Comment by Cojoke-dot** [comment]
> The PR has maintainer approval if there is a timer before the sleep happens. This delay should be controlled by a cvar so that it's easily adjustable on a server-to-server basis.
> 
> An exact time wasn't settled upon but there were suggestions ranging from 10 to 20 minutes.

OOooOoo, It would be nice to have the ability to identify if people have been SSD for a while and they should be moved to cryo. One of my biggest fears of putting a yellow text SSD person in cryo is that they just DC'ed and had been gone for only a few minutes. This also means not dropping and breaking everything in your hand if you happen to lose connection while walking with a jug of chems or something.

**Comment by lzk228** [comment]
> An exact time wasn't settled upon but there were suggestions ranging from 10 to 20 minutes.

wow, it looks pretty long, i'll do 10 for now, someone else can change it later

**Comment by lzk228** [comment]
i've looked into it a little, and it's too complicated for me right now
closing, maybe will come back later
if anyone wants to take it, you are welcome

**Pull request closed without merging**
